src/Legacy/DynamoRevitVersionSelector/ProductLookUp.cs

```diff
@@ -0,0 +1,345 @@
+        IInstalledProduct GetProductFromInstallPath(string path);
+
+        /// <summary>
+        /// Gets installed product from it's name
+        /// </summary>
+        /// <param name="name">Product Name for lookup</param>
+        /// <returns>Installed product</returns>
+        IInstalledProduct GetProductFromProductName(string name);
+
+        /// <summary>
+        /// Gets installed product from it's product id
+        /// </summary>
+        /// <param name="productCode">Product guid string such as 
+        /// {6B5FA6CA-9D69-46CF-B517-1F90C64F7C0B}</param>
+        /// <returns>Installed product</returns>
+        IInstalledProduct GetProductFromProductCode(string productCode);
+
+        /// <summary>
+        /// Returns product name list based on lookup criteria
+        /// </summary>
+        /// <returns>Product name list</returns>
+        IEnumerable<string> GetProductNameList();
+
+        /// <summary>
+        /// Checks if product installation exists at given path
+        /// </summary>
+        /// <param name="installPath">Installation path</param>
+        /// <returns>true for success</returns>
+        bool ExistsAtPath(string installPath);
+
+        /// <summary>
+        /// Returns full path for the core file based on install location
+        /// </summary>
+        /// <param name="installPath">Install location</param>
+        /// <returns>Full path for core file</returns>
+        string GetCoreFilePathFromInstallation(string installPath);
+
+        /// <summary>
+        /// Gets file version info from the given path.
+        /// </summary>
+        /// <param name="filePath">File path</param>
+        /// <returns>Version info as Tuple</returns>
+        Tuple<int, int, int, int> GetVersionInfoFromFile(string filePath);
+
+    }
+
+    /// <summary>
+    /// Represents collection of installed productss
+    /// </summary>
+    public interface IProductCollection
+    {
+        /// <summary>
+        /// Gets list of installed products
+        /// </summary>
+        IEnumerable<IInstalledProduct> Products { get; }
+
+        /// <summary>
+        /// Gets latest product from installation
+        /// </summary>
+        /// <returns>Installed product</returns>
+        IInstalledProduct GetLatestProduct();
+
+        /// <summary>
+        /// Gets all installed product on the system using the given lookUp 
+        /// and initializes itself.
+        /// </summary>
+        /// <param name="lookUp">LookUp interface</param>
+        void LookUpAndInitProducts(IProductLookUp lookUp);
+    }
+
+    /// <summary>
+    /// Implements basic look up algorithm to get product installations using registry keys
+    /// </summary>
+    public class InstalledProductLookUp : IProductLookUp
+    {
+        const string REG_KEY64 = @"SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\";
+        const string REG_KEY32 = @"SOFTWARE\Wow6432Node\Microsoft\Windows\CurrentVersion\Uninstall\";
```

Comment:
Windows installation maked an entry to these registry key for 64bit or 32bit installation. Innosetup installer makes use of 32bit installation whereas msi makes use of 64bit installation.